order-service/src/controllers/order.controller.ts

```diff
@@ -198,6 +199,13 @@ export class OrderController {
                 return;
             }
 
+            // Notify user the order is ready for checkout
+            if (Number(statusId) === 2) {
```

Comment:
This is a business rule and should belong to `OrderService`, not to this controller.

Reply:
Also, it would be cool to have something like this:
```ts
if (Number(statusId) === OrderStatus.WAITING_CHECKOUT) { ... }
```
but some changes would be required:
```ts
// utils/OrderStatus.ts
export default {
    NEW: 0,
    PREPARING: 1,
    WAITING_CHECKOUT: 2,
    WITHDRAWN: 3,
    CANCELED: 4,
};
```

```ts
// entities/order.entity.ts
@Column()
@IsIn(Object.values(OrderStatus)), {
    message: Messages.validation.order_status_dont_exist,
})
public statusId: number;
```

```ts
// services/order.service.ts (line 117)
if (Object.values(OrderStatus).includes(statusId)) { ... }
```

What do you guys think? @eduardosm7 @aacgn @MateusPonciano

Reply:
I believe that's actually a great idea, but I'm afraid we don't have time for it. Thanks for the suggestion :D